include/openthread/platform/radio.h

```diff
@@ -790,6 +814,66 @@ bool otPlatRadioIsCoexEnabled(otInstance *aInstance);
  */
 otError otPlatRadioGetCoexMetrics(otInstance *aInstance, otRadioCoexMetrics *aCoexMetrics);
 
+/**
+ * Get the current time (64 bits width) of the radio chip.
+ *
+ * @returns The current time in microseconds.
+ *
+ */
+uint64_t otPlatRadioGetNow(void);
+
+/**
+ * Enable or disable CSL receiver.
+ *
+ * @param[in]  aInstance     The OpenThread instance structure.
+ * @param[in]  aCslPeriod    CSL period, 0 for disabling CSL.
+ * @param[in]  aExtAddr      The extended source address of the CSL transmitter.
+ *
+ * @retval  OT_ERROR_NOT_SUPPORTED  Radio driver doesn't support CSL.
+ * @retval  OT_ERROR_NONE           Successfully enabled or disabled CSL.
```

Comment:
`otPlatRadioEnableCsl` allows an update for csl period. That is, even if csl has started, we can still call it update the `csl period`. So it's ok to call it both when csl has been enabled or not.
The other errors I can think here:
* Invalid arguments. But I think it's not quite necessary because the mac layer would do the check.
* Other error. When platform API like `nrf_802154_ack_data_set` fails.